src/storage/mod.rs

```diff
@@ -44,6 +49,8 @@ pub enum DatabaseError {
     BinDe(DecodingError),
     Byteorder(::byteorder::Error),
     MagicNmbr,
+    Engine,
+    DataBase,
```

Comment:
Those three error variants are not really saying much with their names. How about `WrongMagicNo` or something like that. And what is `Engine` and `DataBase` supposed to mean?